code/Test_definitions/Geofencing.feature

```diff
@@ -0,0 +1,297 @@
+  Scenario:  Create geofencing subscription with invalid parameter
+    Given the request body is not compliant with the schema "/components/schemas/SubscriptionRequest"
+    When the  request "createSubscription" is sent 
+    Then the response code is 400
+    And the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+
+ @geofencing_subscriptions_12_creation_of_subscription_with_expiry_time_in_past
+  Scenario: Expiry time in past
+    Given a valid subscription request body with expiry time in past
+    When the  request "createSubscription" is sent 
+    Then the response code is 400
+    And the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+	  	
+ @geofencing_subscription_13_creation_with_invalid_eventType
+   Scenario: Subscription creation with invalid event type
+    Given a valid subscription request body 
+    And the request body property "$.types" is set to invalid value  
+    When the request "createSubscription" is sent
+    Then the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+    
+  @geofencing_subscription_14_invalid_protocol
+   Scenario: Subscription creation with invalid protocol
+    Given a valid subscription request body 
+    When the request "createSubscription" is sent
+    And "$.types"="org.camaraproject.geofencing-subscriptions.v0.area-entered"
+    And "$.protocol"<>"HTTP"
+    And "$.config.subscriptionDetail.phoneNumber" is set with with provided phoneNumber
+    And "$.sink" is set to provided callbackUrl
+    Then the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_PROTOCOL"
+    And the response property "$.message" contains a user friendly text
+
+  @geofencing_subscription_15_invalid_credential
+  Scenario: Subscription creation with invalid credential
+    Given a valid subscription request body 
+    When the request "createSubscription" is sent
+    And "$.types"="org.camaraproject.geofencing-subscriptions.v0.area-entered"
+    And "$.protocol"="HTTP"
+    And "$.config.subscriptionDetail.phoneNumber" is set with with provided phoneNumber
+    And "$.sink" is set to provided callbackUrl
+    And "$.sinkCredential.credentialType" <> "ACCESSTOKEN"
+    And "$.sinkCredential.accessTokenType" = "bearer"
+    And "$.sinkCredential.accessToken" is valued with a valid value
+    And "$.sinkCredential.accessTokenExpiresUtc" is valued with a valid value
+    Then the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_CREDENTIAL"
+    And the response property "$.message" contains a user friendly text
+
+  @geofencing_subscription_16_invalid_token
```

Comment:
As "bearer" is in the only value accepted in the schema (enum), some implementations may reject the request with INVALID_ARGUMENT because the request body is not compliant with the schema, so probably both options are acceptable. For the basic test plan I think we can keep both